crates/component/src/wrappers/io.rs

```diff
@@ -119,6 +119,26 @@ impl Default for StdioStream<'_> {
     }
 }
 
+/// Similar to [`crate::wasi::io::poll::poll`], but polls all `pollables` until they are all ready.
+///
+/// Poll for completion on a set of pollables.
+///
+/// This function takes a list of pollables, which identify I/O sources of interest, and waits until all of the events are ready for I/O.
+pub fn poll_all(pollables: &[&crate::wasi::io::poll::Pollable]) {
+    let mut pollables = pollables.to_vec();
+    while !pollables.is_empty() {
+        let ready_indices = crate::wasi::io::poll::poll(&pollables);
+        pollables = pollables
+            .into_iter()
+            .enumerate()
+            .filter_map(|(i, p)| {
+                // Retain the pollables that didn't return ready
+                ready_indices.contains(&(i as u32)).then_some(p)
```

Comment:
```suggestion
                (!ready_indices.contains(&(i as u32))).then_some(p)
```
This would need to return `Some` for all pollables that are *not* ready, right?